tasks/Task.cpp

```diff
@@ -13,37 +17,214 @@ Task::~Task()
 {
 }
 
+bool Task::configureHook()
+{
+    delete m_driver;
+    delete m_state_machine;
```

Comment:
I can't find the reference in the standard by I don't think you can expect these to be null-initialized

Reply:
Indeed not, but they were initialized in the attribute declaration. However, they were double-delete codepaths possible. I'm now setting them to nullptr after the delete.